soccer/gameplay/plays/restarts/basic_offensive_indirect.py

```diff
@@ -0,0 +1,131 @@
+import main
+import robocup
+import behavior
+import constants
+
+import standard_play
+import skills.line_kick
+import tactics.coordinated_pass
+import enum
+import skills.move
+import random
+
+# A basic play for the Offensive Indect kick 
+# Chooses to pass to a close or far robot
+# If it chooses the far bot, it will chip
+# If the ball is in the thrid quarter of the field,
+# The robots will move up
+# If the ball is in the fourth quearter of the field,
+# The robots will move to points in a horizontal line with the ball
+
+class BasicOffensiveIndirect(standard_play.StandardPlay):
+
+    class State(enum.Enum):
+        move = 1 # Move recievers to proper postions
+        kick = 2 # Kick the ball to one of the recievers
+
+    def __init__(self, indirect=None):
+        super().__init__(continuous=True)
+
+        for s in BasicOffensiveIndirect.State:
+            self.add_state(s, behavior.Behavior.State.running)
+
+        self.points = [] # The list that holds the receive points
+
+        self.chip = False # Determines wether to chip or not
+
+        self.receive_pt = None
+
+        self.add_transition(behavior.Behavior.State.start,
+                            BasicOffensiveIndirect.State.move, lambda: True,
+                            'immediately')
+
+        self.add_transition(
+            BasicOffensiveIndirect.State.move,
+            BasicOffensiveIndirect.State.kick, lambda: self.subbehavior_with_name(
+                'move to point 2').state == behavior.Behavior.State.completed, 'kick') # Once the receivers are in position
+
+        self.add_transition(
+            BasicOffensiveIndirect.State.kick,
+            behavior.Behavior.State.completed,
+            lambda: self.subbehavior_with_name('pass').state == behavior.Behavior.State.completed and self.subbehavior_with_name('pass').state != tactics.
+            coordinated_pass.CoordinatedPass.State.timeout,
+            # Keep trying pass until timeout
+            'pass completes') # The pass is complete
+    
+    @classmethod
+    def score(cls):
+        gs = main.game_state()
+        return 10
+        # return 0 if behavior.Behavior.State.running or (
+        #     gs.is_ready_state() and
+        #     gs.is_our_indirect_kick()) else float("inf")
+
+    @classmethod
+    def is_restart(cls):
+        return True
+
+    def calc_pass_point(self): # Determines the position of the recievers and where to pass
+        ball = main.ball().pos # position of ball
+        num = random.randint(0,1) # a random int to choose which receiver to pass to
+        if (num == 1):
+            self.chip = True # If the reciever is the second one, it will be far away and require a chip
```

Comment:
You don't have to comment every line lmao. It's in there, so you don't have to take it out, but for future reference, commenting every line is pretty extra and not always helpful